### PR TITLE
rehearse: compress CONFIG_SPEC

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR updates the `pj-rehearse` tool to compress and base64 encode the
`CONFIG_SPEC` env var in order to allow rehearsal of jobs have a
`CONFIG_SPEC` larger than ~128KB (max linux arglength).

### CODE REVIEW

File: pkg/rehearse/jobs_test.go [hunk 1]
```diff
@@ -648,6 +665,16 @@ func TestExecuteJobsPositive(t *testing.T) {
...
+	compressedJob1, err := gzip.CompressString(testingCiOpCfgJob1YAML)
+	if err != nil {
+		t.Fatalf("Failed to compress config: %v", err)
+	}
+	job1Cfg := base64.StdEncoding.EncodeToString([]byte(compressedJob1))
+	compressedJob2, err := gzip.CompressString(testingCiOpCfgJob2YAML)
+	if err != nil {
+		t.Fatalf("Failed to compress config: %v", err)
+	}
+	job2Cfg := base64.StdEncoding.EncodeToString([]byte(compressedJob2))
```
Comment (alvaroaleman) [inline pkg/rehearse/jobs_test.go:677]: Can we keep one uncompressed to make sure that continues to work?

**Comment by AlexNPavel** [comment]
I've updated the PR to handle the `base64` encode as part of the compress string function and added a unit test loading compressed config from env.

File: pkg/rehearse/jobs_test.go (hunk 1 above)
Comment (AlexNPavel) [reply pkg/rehearse/jobs_test.go]: That variable is the expected config_spec output for that function, so it will always be compressed.

**Pull request merged**